### ISSUE TITLE
Close Preview Should Not Close Program

### ISSUE THREAD

**Issue by KitchM** [oferkv/phototonic#264]
I just stumbled upon your program and am very impressed.  Where do I contribute?

The one annoyance I have found is that whenever I double-click on an image, it opens a new window which closes the program when exiting out of the new window on the top right.  I would like to see it only close the preview window.

Debian 10 with Xfce 4.12

In any case, good job!

**Comment by barjac** [comment]
I agree, I remember commenting on this a long time ago.

**Comment by sandsmark** [comment]
Not sure I understand this, it doesn't open any new window and doesn't show any close button on the top right?

**Comment by KitchM** [comment]
Please see above:

>  ",,,,whenever I double-click on an image, it opens a new window..."

It displays the image in a new, full-screen window.  This is evidently the default action.

> "....which closes the program when exiting out of the new window on the top right"

This means that when the full-screen display window is closed (using the top right X button), the whole program in the background is closed  as well.  Everything simply shuts down.

**Comment by oferkv** [comment]
Thanks, KitchM.
I guess you are referring to a UX in which the thumbnail viewer and the image viewer are different windows (Like the IrfanView Windows application for example).
Phototonic is more of a ACDsee clone, in which both viewers are implemented in the same window.
This of course could be implemented, as an option.

**Comment by KitchM** [comment]
Well, the UX always is determined by the UI.  Poor UI design always creates a poor UX.

I cannot remember any time that I have seen this sort of action in any program, nor remember any customer mentioning such action to me.

Any time a program needs to open another window, two things come to mind.  First, why didn't it open in a new tab instead?  Second, how does the second window's functions get tangled up with the main application?  The first is always preferred and the second should never be.

I am not convinced that any repair for this odd behavor should be an "option".  Spawning a new window has nothing to do with the first once the spawning is completed.  I can't imagine any possible benefit for this odd behavior.

**Comment by oferkv** [comment]
> Well, the UX always is determined by the UI. Poor UI design always creates a poor UX.

Tell that to ACDsee v3.0  :smile: 

Anyway, feel free to implement it if you wish (as an option).

**Comment by KitchM** [comment]
That's not how you get donations.

**Comment by sandsmark** [comment]
> It displays the image in a new, full-screen window. This is evidently the default action.

No, it is the same window.

I think the default button to go back to the thumbnail view is `Esc`.

**Comment by KitchM** [comment]
I'm not saying you are not correct, but it is odd behavior none the less.  Besides which, where are the menu and title bars and such?  This is more like calling up a fresh and empty window.

I am using LightDM.

**Comment by sandsmark** [comment]
yeah, I agree that it's not obvious that it's a new window, but it is consisten with the fullscreen mode i things like chromium, gwenview, and a lot of other apps.

but I agree that it is a better UX if it's a separate window, and it shouldn't be fairly trivial to fix (hopefully it is enough to not set a parent widget for the fullscreen view, that should make qt show it in a separate window), so I'll look into it later.

**Comment by KitchM** [comment]
Since I use Linux, some other software is unfamiliar to me.  However, I did check Chromium since I rarely use it, and I found that it does **not** open in new windows but rather in new tabs, as any program should.  A cross-platform example of good programming would be GIMP.  Only by exiting out of the main window will the program be closed.

I hope you meant that it **would** be trivial to fix.  If so, I would agree.

**Comment by sandsmark** [comment]
> However, I did check Chromium since I rarely use it, and I found that it does not open in new windows but rather in new tabs

Yeah, I typoed a lot in the comment, so maybe wasn't clear, but I meant fullscreen mode.

> I hope you meant that it would be trivial to fix. If so, I would agree.

Yes, or rather "should", since I didn't look into the code.
